packages/@lwc/engine-core/src/framework/vm.ts

```diff
@@ -150,6 +145,9 @@ export interface VM<N = HostNode, E = HostElement> {
     ) => any;
 }
 
+let profilerEnabled = false;
+trackProfilerState((t) => (profilerEnabled = t));
```

Comment:
Are you doing this hoisting here for performance? Why not just do a property check on the profiler?

Reply:
Yes. I found that the best metrics were regressed. 
The base does no work in prod mode since the compiler removes the node env check block.

If we do a property check, now we're doing a property lookup on an object and a boolean check. 

The boolean check is extremely cheap comparatively. Hence the current approach

Reply:
How do I kick off a best run for this PR?

Reply:
Needs to be a branch rather than a fork (due to security key management)